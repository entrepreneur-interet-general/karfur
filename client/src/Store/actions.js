export const TOGGLE_LANGUE = 'TOGGLE_LANGUE';

export const TOGGLE_TTS = 'TOGGLE_TTS';

