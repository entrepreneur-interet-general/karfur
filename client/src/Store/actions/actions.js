export const TOGGLE_LANGUE = 'TOGGLE_LANGUE';
export const TOGGLE_LANG_MODAL = 'TOGGLE_LANG_MODAL';

export const TOGGLE_TTS = 'TOGGLE_TTS';

export const GET_LANGUES = 'GET_LANGUES';

export const get_langues = () => {
  return {
    type: GET_LANGUES
  };
};