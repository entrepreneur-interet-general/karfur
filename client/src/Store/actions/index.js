export {
  fetch_langues,
  toggle_lang_modal,
  toggle_langue } from './langues'

export { fetch_dispositifs } from './dispositifs'
export { fetch_structures } from './structures'

export { fetch_user, update_user } from './user'

export { toggleTTS, toggleSpinner } from './tts'