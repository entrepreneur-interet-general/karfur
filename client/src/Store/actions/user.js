import * as actionTypes from './actionTypes';
import API from '../../utils/API';

const set_user = (value) => {
  return {
    type: actionTypes.SET_USER,
    value: value
  }
};

export const fetch_user = () => {
  return dispatch => {
    if(API.isAuth()){
      return API.get_user_info().then(data => {
        return  dispatch(set_user(data.data.data));
      })
    }else{ return false; }
  };
};

export const update_user = value => ({ type: actionTypes.UPDATE_USER, value: value });