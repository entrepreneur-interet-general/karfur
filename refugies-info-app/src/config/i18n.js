export const fallback = "fr";
export const supportedLocales = {
  fr: {
    name: "Français",
    translationFileLoader: () => require("../lang/fr.json"),
    momentLocaleLoader: () => import("moment/locale/fr"),
  },
  en: {
    name: "English",
    translationFileLoader: () => require("../lang/en.json"),
    // en is default locale in Moment
    momentLocaleLoader: () => Promise.resolve(),
  },
  ar: {
    name: "عربي",
    translationFileLoader: () => require("../lang/ar.json"),
    momentLocaleLoader: () => import("moment/locale/ar"),
  },
};
