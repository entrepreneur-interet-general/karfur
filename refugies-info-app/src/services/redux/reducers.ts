import { Action, combineReducers } from "redux";
import {
  loadingStatusReducer,
  LoadingStatusState,
  initialLoadingStateFactory,
} from "./LoadingStatus/loadingStatus.reducer";

const appReducer = combineReducers({
  loadingStatus: loadingStatusReducer,
  //   languages : languagesReducer
});

export interface RootState {
  loadingStatus: LoadingStatusState;
}

type RootReducer = ReturnType<typeof appReducer>;

export const initialRootStateFactory = (): RootState => ({
  loadingStatus: initialLoadingStateFactory(),
});

export const rootReducer = (
  state: RootState | undefined,
  action: Action<any>
): RootReducer =>
  // @ts-ignore
  appReducer(state, action);
