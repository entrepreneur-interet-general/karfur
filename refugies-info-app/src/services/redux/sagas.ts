import { SagaIterator } from "redux-saga";
import { fork } from "redux-saga/effects";
import usersSaga from "./User/user.saga";

export function* rootSaga(): SagaIterator {
  yield fork(usersSaga);
}
