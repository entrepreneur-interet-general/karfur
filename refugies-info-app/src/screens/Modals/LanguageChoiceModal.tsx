import { StyleSheet, View } from "react-native";
import Modal from "react-native-modal";
import React from "react";
import styled from "styled-components/native";
import { useDispatch } from "react-redux";
import i18n, { t } from "../../services/i18n";
import { saveSelectedLanguageActionCreator } from "../../services/redux/User/user.actions";
import { theme } from "../../theme";
import {
  StyledTextNormal,
  StyledTextNormalBold,
} from "../../components/StyledText";
import { LanguageDetailsButton } from "../../components/Language/LanguageDetailsButton";
import { getSelectedLanguageFromI18nCode } from "../../libs/language";
import { activatedLanguages } from "../../data/languagesData";
import { RowContainer } from "../../components/BasicComponents";
import { Flag } from "../../components/Language/Flag";

interface Props {
  isModalVisible: boolean;
  toggleModal: () => void;
  selectedLanguageI18nCode: string | null;
}

const styles = StyleSheet.create({
  view: {
    justifyContent: "flex-end",
    margin: 0,
  },
});
const MainContainer = styled.TouchableOpacity`
  background: ${theme.colors.white};
  padding: ${theme.margin * 2}px;
  margin-vertical: ${theme.margin}px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  flex-wrap: wrap;
`;

const StyledTextBold = styled(StyledTextNormalBold)`
  text-align: left;
  margin-left: ${theme.margin}px;
  color: ${(props: { isSelected: boolean }) =>
    props.isSelected ? theme.colors.white : theme.colors.black};
`;

const StyledText = styled(StyledTextNormal)`
  text-align: left;
  color: ${(props: { isSelected: boolean }) =>
    props.isSelected ? theme.colors.white : theme.colors.black};
`;

const ModalView = styled.View`
  background-color: ${theme.colors.lightGrey};
  display: flex;
  padding-top: ${theme.margin}px;
  padding-left: ${theme.margin * 3}px;
  padding-right: ${theme.margin * 3}px;
`;

const TitleContainer = styled(StyledTextNormal)`
  margin-bottom: ${theme.margin}px;
`;

const OtherLanguagesContainer = styled.View`
  background-color: ${theme.colors.white};
  border-radius: ${theme.radius * 2}px;
  margin-top: ${theme.margin * 3}px;
  margin-bottom: ${theme.margin * 3}px;
`;

const Separator = styled.View`
  height: 1px;
  background-color: ${theme.colors.grey};
  margin-left: ${theme.margin * 2}px;
  margin-right: ${theme.margin * 2}px;
`;
export const LanguageChoiceModal = (props: Props) => {
  const dispatch = useDispatch();

  const selectedLanguage = getSelectedLanguageFromI18nCode(
    props.selectedLanguageI18nCode
  );

  const otherLanguages = activatedLanguages.filter(
    (langue) => langue.i18nCode !== props.selectedLanguageI18nCode
  );

  const changeLanguage = (ln: string) => {
    i18n.changeLanguage(ln);
    dispatch(saveSelectedLanguageActionCreator(ln));
    props.toggleModal();
  };
  return (
    <Modal
      isVisible={props.isModalVisible}
      style={styles.view}
      onBackdropPress={props.toggleModal}
    >
      <ModalView>
        <TitleContainer>
          {t("Langue app", "Langue de l'application")}
        </TitleContainer>
        <LanguageDetailsButton
          langueFr={selectedLanguage.langueFr}
          langueLoc={selectedLanguage.langueLoc}
          onPress={() => {}}
          isSelected={true}
        />
        <OtherLanguagesContainer>
          {otherLanguages.map((language, index) => {
            return (
              <View key={index}>
                <MainContainer
                  onPress={() => changeLanguage(language.i18nCode)}
                  testID={"test-language-button-" + language.langueFr}
                  key={index}
                >
                  <RowContainer>
                    <Flag langueFr={language.langueFr} />
                    <StyledTextBold>{language.langueFr + " - "}</StyledTextBold>
                    <StyledText>{language.langueLoc}</StyledText>
                  </RowContainer>
                </MainContainer>
                {index !== otherLanguages.length - 1 && <Separator />}
              </View>
            );
          })}
        </OtherLanguagesContainer>
      </ModalView>
    </Modal>
  );
};
