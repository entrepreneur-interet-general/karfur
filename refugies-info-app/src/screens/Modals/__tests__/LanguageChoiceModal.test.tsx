import { wrapWithProvidersAndRender } from "../../../jest/wrapWithProvidersAndRender";
import { initialRootStateFactory } from "../../../services/redux/reducers";
import { fetchLanguagesActionCreator } from "../../../services/redux/Languages/languages.actions";
import { fireEvent, act } from "react-native-testing-library";
import { saveSelectedLanguageActionCreator } from "../../../services/redux/User/user.actions";
import i18n from "../../../services/i18n";
import { mockedLanguageData } from "../../../jest/__fixtures__/languages";
import { LanguageChoiceModal } from "../LanguageChoiceModal";

jest.mock("../../../services/redux/Languages/languages.actions", () => {
  const actions = jest.requireActual(
    "../../../services/redux/Languages/languages.actions"
  );
  return {
    fetchLanguagesActionCreator: jest.fn(actions.fetchLanguagesActionCreator),
  };
});

jest.mock("../../../services/i18n", () => ({
  __esModule: true, // this property makes it work
  default: { changeLanguage: jest.fn(), isRTL: jest.fn() },
  t: jest.fn(),
}));

jest.mock("../../../services/redux/User/user.actions", () => {
  const actions = jest.requireActual(
    "../../../services/redux/User/user.actions"
  );
  return {
    saveSelectedLanguageActionCreator: jest.fn(
      actions.saveSelectedLanguageActionCreator
    ),
  };
});

describe("LanguageChoiceModal", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const toggleModal = jest.fn();
    const component = wrapWithProvidersAndRender({
      Component: LanguageChoiceModal,
      reduxState: {
        ...initialRootStateFactory(),
        languages: { availableLanguages: mockedLanguageData },
      },
      compProps: { toggleModal },
    });
    expect(fetchLanguagesActionCreator).toHaveBeenCalledWith();
    expect(component).toMatchSnapshot();
    const Button = component.getByTestId("test-language-button-Anglais");
    act(() => {
      fireEvent.press(Button);
    });
    expect(i18n.changeLanguage).toHaveBeenCalledWith("en");
    expect(saveSelectedLanguageActionCreator).toHaveBeenCalledWith("en");
    expect(toggleModal).toHaveBeenCalledWith();
  });
});
