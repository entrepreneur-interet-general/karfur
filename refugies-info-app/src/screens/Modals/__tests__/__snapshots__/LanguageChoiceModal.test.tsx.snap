// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageChoiceModal should render correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[MockFunction]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-end",
        "margin": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F6F6F6",
            "display": "flex",
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 8,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "circularBook",
              "fontSize": 19,
              "marginBottom": 8,
              "textAlign": "left",
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0421B1",
            "borderRadius": 12,
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-between",
            "marginVertical": 8,
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
        testID="test-language-button-Français"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "display": "flex",
                  "height": 17,
                  "justifyContent": "center",
                  "marginBottom": 4,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 4,
                  "width": 22,
                },
              ]
            }
          >
            <react-native-svg
              height={15}
              width={20}
            />
          </View>
          <Text
            isSelected={true}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "circularBold",
                  "fontSize": 16,
                  "marginLeft": 8,
                  "textAlign": "left",
                },
              ]
            }
          >
            Français
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 12,
              "marginBottom": 24,
              "marginTop": 24,
            },
          ]
        }
      >
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "display": "flex",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginVertical": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="test-language-button-Français"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <react-native-svg
                height={15}
                width={20}
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212121",
                      "fontFamily": "circularBold",
                      "fontSize": 16,
                      "marginLeft": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Français
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EDEBEB",
                  "height": 1,
                  "marginLeft": 16,
                  "marginRight": 16,
                },
              ]
            }
          />
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "display": "flex",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginVertical": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="test-language-button-Anglais"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <react-native-svg
                height={15}
                width={20}
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212121",
                      "fontFamily": "circularBold",
                      "fontSize": 16,
                      "marginLeft": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Anglais
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "circularBold",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                   - 
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#212121",
                        "fontFamily": "circularBook",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  English
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EDEBEB",
                  "height": 1,
                  "marginLeft": 16,
                  "marginRight": 16,
                },
              ]
            }
          />
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "display": "flex",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginVertical": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="test-language-button-Pachto"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <react-native-svg
                height={15}
                width={20}
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212121",
                      "fontFamily": "circularBold",
                      "fontSize": 16,
                      "marginLeft": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Pachto
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "circularBold",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                   - 
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#212121",
                        "fontFamily": "circularBook",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  پښتو
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EDEBEB",
                  "height": 1,
                  "marginLeft": 16,
                  "marginRight": 16,
                },
              ]
            }
          />
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "display": "flex",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginVertical": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="test-language-button-Persan/Dari"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <react-native-svg
                height={15}
                width={20}
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212121",
                      "fontFamily": "circularBold",
                      "fontSize": 16,
                      "marginLeft": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Persan/Dari
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "circularBold",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                   - 
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#212121",
                        "fontFamily": "circularBook",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  فارسی/ دری
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EDEBEB",
                  "height": 1,
                  "marginLeft": 16,
                  "marginRight": 16,
                },
              ]
            }
          />
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "display": "flex",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginVertical": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="test-language-button-Tigrinya"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <react-native-svg
                height={15}
                width={20}
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212121",
                      "fontFamily": "circularBold",
                      "fontSize": 16,
                      "marginLeft": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Tigrinya
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "circularBold",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                   - 
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#212121",
                        "fontFamily": "circularBook",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  ትግርኛ
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EDEBEB",
                  "height": 1,
                  "marginLeft": 16,
                  "marginRight": 16,
                },
              ]
            }
          />
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "display": "flex",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginVertical": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="test-language-button-Russe"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <react-native-svg
                height={15}
                width={20}
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212121",
                      "fontFamily": "circularBold",
                      "fontSize": 16,
                      "marginLeft": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Russe
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "circularBold",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                   - 
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#212121",
                        "fontFamily": "circularBook",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  Русский
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#EDEBEB",
                  "height": 1,
                  "marginLeft": 16,
                  "marginRight": 16,
                },
              ]
            }
          />
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "display": "flex",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "marginVertical": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
            testID="test-language-button-Arabe"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <react-native-svg
                height={15}
                width={20}
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#212121",
                      "fontFamily": "circularBold",
                      "fontSize": 16,
                      "marginLeft": 8,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Arabe
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "circularBold",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                   - 
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#212121",
                        "fontFamily": "circularBook",
                        "fontSize": 16,
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  العربية
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
